src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Change.cs

```diff
@@ -1,14 +1,170 @@
-using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
 
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
 
-public class Change<TState>
+public abstract class Change<TEntity>
```

Comment:
Had you considered adding an additional `TId` generic param to this base class?  Then you could have:

```
    public TId? CurrentStateId { get; set; }

    public TId? PreviousStateId { get; set; }
```

on this base class and not have to add it to each subclass?

Reply:
I originally did model it like this, but it's not a correct generalisation for all the change types. For example, location and filter option changes are using owned types and don't use a single ID, instead they reference meta / option IDs like the link tables do. Consequently, this isn't really possible.

Reply:
Gotcha that makes sense, thanks!